server/src/main/java/org/elasticsearch/repositories/RepositoryException.java

```diff
@@ -20,12 +21,12 @@
 public class RepositoryException extends ElasticsearchException {
     private final String repository;
 
-    public RepositoryException(String repository, String msg) {
-        this(repository, msg, null);
+    public RepositoryException(String repository, String msg, Object... args) {
+        this(repository, msg, (Throwable)null, args);
     }
 
-    public RepositoryException(String repository, String msg, Throwable cause) {
-        super("[" + (repository == null ? "_na" : repository) + "] " + msg, cause);
+    public RepositoryException(String repository, String msg, Throwable cause, Object... args) {
+        super("[" + (repository == null ? "_na" : repository) + "] " + LoggerMessageFormat.format(msg, args), cause);
```

Comment:
Here's the change to `RepositoryException` that I mentioned in the other comment chain, which is based on a similar constructor in `ElasticsearchException` [here](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/ElasticsearchException.java#L142). Is that the desired change to avoid repeating `MessageFormat.format` calls?